sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -947,7 +947,10 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
     withTable(table) {
       sql(s"create table $table (a array<string collate utf8_lcase>) using parquet")
       sql(s"insert into $table values (array('aaa')), (array('AAA'))")
-      checkAnswer(sql(s"select distinct a from $table"), Seq(Row(Seq("aaa"))))
+      val result = sql(s"select distinct a from $table").collect()
+      assert(result.length === 1)
+      val data = result.head.get(0).asInstanceOf[scala.collection.mutable.ArraySeq[String]]
+      assert(data === Array("aaa") || data === Array("AAA"))
```

Comment:
so the test becomes flaky? Reading the test query I think both `aaa` and `AAA` are correct result. cc @stefankandic

Reply:
Yes, it becomes flaky. The returned value is not non-deterministic.

Reply:
You mean just non-deterministic, I guess.